dotcom-rendering/src/web/components/FetchCommentCounts.importable.tsx

```diff
@@ -0,0 +1,173 @@
+import { useState } from 'react';
+import { renderToString } from 'react-dom/server';
+import { formatCount } from '../lib/formatCount';
+import { useApi } from '../lib/useApi';
+import { CardCommentCount } from './CardCommentCount';
+
+type Props = {
+	repeat?: boolean;
+};
+
+type RawCountType = {
+	id: string;
+	count: number;
+};
+
+type EnhancedCountType = {
+	id: string;
+	long: string;
+	short: string;
+	format: ArticleFormat;
+};
+
+type RenderedCountType = {
+	id: string;
+	html: string;
+	short: string;
+};
+
+/**
+ * @description
+ * Reads the dom looking for markers that contain the information needed to render the
+ * count
+ */
+function extractMarkers() {
+	const markers: { discussionId: string; format: ArticleFormat }[] = [];
+	document
+		.querySelectorAll('[data-name="comment-count-marker"]')
+		.forEach((element: Element) => {
+			if (element instanceof HTMLElement) {
+				const { discussionid: discussionId, format } = element.dataset;
+				if (discussionId && format)
+					markers.push({ discussionId, format: JSON.parse(format) });
+			}
+		});
+	return markers;
+}
+
+/**
+ * @description
+ * Creates the url used to fetch the count data
+ */
+function buildUrl() {
```

Comment:
I’d rather this function take in an array of markers and return a `string`, making it purely functional and testable.

I don’t see why we would need to collect the markers inside the body function.

Reply:
Done. I had originally added the check for markers inside here as I wanted it to be 'fresh' but that's the wrong approach. I've changed things now to - hopefully - be clearer. Let me know what you think